Fixed switching between speaker and headset

natanavra opened the pull request:
Switching between speaker and headset didn't work always.
Now it does, for me at least. 
Tested on iPhone 5s iOS 7

wobbals commented on 7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m:
```diff
@@ -1062,10 +1062,8 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
     //ios 8.0 complains about Deactivating an audio session that has running
     // I/O. All I/O should be stopped or paused prior to deactivating the audio
     // session. Looks like we can get away by not using the setActive call
-    if (SYSTEM_VERSION_LESS_THAN_OR_EQUAL_TO(@"7.0")) {
```
Is there any reason to drop the iOS 7 guards?

natanavra commented:
The moment I've moved the 'setActive:' to the end and removed the 'of' so it executes on iOS 7 and above it started working.
I'm not familiar enough with the AVAudioSession class, to tell you what's behind this - but this seema to do the trick.

wobbals commented on 7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m:
```diff
@@ -1094,6 +1085,20 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
                                         error:&err];
     }
     
+    if([AUDIO_DEVICE_HEADSET isEqualToString: desiredAudioRoute]) {
```
This seems like the only actual change other than the reordering of the `setActive:` calls. Re-add the removed iOS7 guards around `setActive:`, you can keep the reordering, otherwise lgtm.

natanavra replied:
See my comment on the pull request.
This was tested on iPhone 5 iOS 7.
Meaning these blocked the code from executing and it did not work.

So it's my bad. The guards and the re-ordering need to be there.

On Mon, Jun 15, 2015 at 9:13 PM, Charley Robinson <notifications@github.com>
wrote:

> In 7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m
> <https://github.com/opentok/opentok-ios-sdk-samples/pull/97#discussion_r32449490>
> :
>
> > @@ -1094,6 +1085,20 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
> >                                          error:&err];
> >      }
> >
> > +    if([AUDIO_DEVICE_HEADSET isEqualToString: desiredAudioRoute]) {
>
> This seems like the only actual change other than the reordering of the
> setActive: calls. Re-add the removed iOS7 guards around setActive:, you
> can keep the reordering, otherwise lgtm.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opentok/opentok-ios-sdk-samples/pull/97/files#r32449490>
> .
>

wobbals commented:
The issues fixed here were resolved in a few later pull requests by @bsrao. Closing.
